ecs-init/docker/docker.go

```diff
@@ -571,6 +576,25 @@ func nvidiaGPUDevicesPresent() bool {
 	return true
 }
 
+// nvidiaGPUDevicesPresentWithRetries checks if NVIDIA GPU devices are present in the instance. It retries if NVIDIA
+// GPU devices are not yet present every `nvidiaGPUDevicesPresentRetryTime` interval of time up to a maximum of
+// `nvidiaGPUDevicesPresentMaxRetries` retries.
+func nvidiaGPUDevicesPresentWithRetries() bool {
```

Comment:
please add a unit test for this. i think you can mock nvidiaGPUDevicesPresent()

Reply:
Added in the most recent commit of this pull request - thanks!